File: compy_cli/test_dir/cpp/src/readme_examples/union.cpp
#include "readme_examples/union.h"
#include "compy_union.h"
#include "compy_util/print.h"
#include "py_list.h"
#include "py_str.h"

void union_example() {
    Uni<int, double, PyList<int>> int_float_or_list(3.14);
    if (int_float_or_list.isinst<double>()) {
        double val = int_float_or_list.ug<double>();
        print(val);
    }
    Uni<int, std::monostate> b(std::monostate{});
    if (b.is_none()) {
        print(PyStr("b is None"));
    }
}

File: compy_cli/test_dir/python/src/readme_examples/union.py
from compy_python import Uni, ug, isinst, is_none


def union_example():
    int_float_or_list: Uni[int, float, list[int]] = Uni(3.14)
    if isinst(int_float_or_list, float):
        val: float = ug(int_float_or_list, float)
        print(val)
    # Union with None
    b: Uni[int, None] = Uni(None)
    if is_none(b):
        print("b is None")
